test/AutoRest.TestServer.Tests/Infrastructure/TestServerLowLevelTestBase.cs

```diff
@@ -81,7 +81,7 @@ private async Task Test(string scenario, Func<Uri, Task> test, bool ignoreScenar
         private static string GetScenarioName()
         {
             var testName = TestContext.CurrentContext.Test.Name;
-            var indexOfUnderscore = testName.IndexOfAny(new[]{'_', '('});
+            var indexOfUnderscore = testName.LastIndexOfAny(new[]{'_', '('});
```

Comment:
Previously we have cases to use multiple tests to test different cases of the same scenario like `DpgNewBodyType_JSON` and `DpgNewBodyType_JPEG` will both count coverage for `DpgNewBodyType`. Therefore, we have this logic to drop the part after the underscore.

In the updated testserver, however, the scenario name itself can have underscores like `DpgAddOptionalInput_OptionalParam`. Therefore, I updated here to drop the part after the last underscore and we can append `_Test` (for instance, use `DpgAddOptionalInput_OptionalParam_Test`) to the test name if the test name contains underscores.